Assets/Scripts/Source/Game/Combat/Attack.cs

```diff
@@ -8,8 +8,25 @@ public virtual int Damage
         private set;
     }
 
-    public Attack(Character actor, int damage, Action onAttackHandler = null, params Character[] targets) : base(actor, onAttackHandler, targets)
+    public virtual int Bonus
+    {
+        get;
+        private set;
+    }
+
+    public Attack(Character actor, int damage, int bonus, Action onAttackHandler = null, params Character[] targets) : base(actor, onAttackHandler, targets)
     {
         this.Damage = damage;
+        this.Bonus = bonus;
+    }
+
+    public override void AddBonus()
```

Comment:
Should this be stateless? 
```csharp 
int GetBonus(int multiplier)
{
    return Damage + Bonus * multiplier;
}
```
Seems that needed to reset this could become confusing

Reply:
This way is simpler so that the stat manager doesn't have to worry about whether or not the bonus was hit. Reset actually isn't ever called since each attack is created new and only gets a bonus in the combat manager routine. If we do extend this to a combo system of sorts we can add bonus each correct hit and reset if failed.